File: baekjoon/8주차/9465-스티커/제문/9465.kt
package baekjoon

import java.lang.StringBuilder
import java.util.*
import kotlin.math.max

private val out = StringBuilder()
private val br = System.`in`.bufferedReader()

private fun readLine() = br.readLine()!!
private fun readInt() = br.readLine().toInt()
private fun readInts() = br.readLine().split(" ").map { it.toInt() }

fun main() {
    repeat(readInt()) {
        val n = readInt()
        val stickers = Array(n + 1) { IntArray(2) } // 0 ~ n행, 2열

        for (i in 0 until 2) {
            val st = StringTokenizer(readLine(), " ")

            for (j in 1..st.countTokens()) {
                stickers[j][i] = st.nextToken().toInt()
            }
        }

        fun solve(): Int {
            // 2행부터 n행까지
            // 각각마다 2열 계산
            for (i in 2..n) {
                for (j in 0 until 2) {
                    val maxValue = maxOf(stickers[i-2][0], stickers[i-2][1], stickers[i-1][j.xor(1)])
                    stickers[i][j] += maxValue
                }
            }
            return max(stickers[n][0], stickers[n][1])
        }
        out.appendln(solve())
    }
    print(out)
}

File: baekjoon/8주차/9465-스티커/정우/9465.py
import sys

for _ in range(int(sys.stdin.readline())):
    n = int(sys.stdin.readline())
    a1 = list(map(int, sys.stdin.readline().split()))
    a2 = list(map(int, sys.stdin.readline().split()))
    dp = [[0]*n for _ in range(3)]
    dp[0][0] = a1[0]
    dp[1][0] = a2[0]
    for i in range(1, n):
        dp[0][i] = max(dp[1][i-1], dp[2][i-1])+a1[i]
        dp[1][i] = max(dp[0][i-1], dp[2][i-1])+a2[i]
        dp[2][i] = max(dp[0][i-1], dp[1][i-1])
    print(max([dp[0][n-1], dp[1][n-1], dp[2][n-1]]))